src/jdk.jlink/share/classes/jdk/tools/jlink/builder/DefaultImageBuilder.java

```diff
@@ -567,7 +572,7 @@ public static ExecutableImage getExecutableImage(Path root) {
         Path binDir = root.resolve(BIN_DIRNAME);
         if (Files.exists(binDir.resolve("java")) ||
             Files.exists(binDir.resolve("java.exe"))) {
-            return new DefaultExecutableImage(root, retrieveModules(root), Platform.UNKNOWN);
+            return new DefaultExecutableImage(root, retrieveModules(root), Platform.runtime());
```

Comment:
Hello Roger, I couldn't find any usages of this (existing) `public static ExecutableImage getExecutableImage(Path root)` in the code. Maybe we could remove it?

Reply:
If we do remove this method, the `private static Set<String> retrieveModules(Path root)` method in this same class will become unused too, so can be removed too.